src/layouts/index.js

```diff
@@ -28,13 +30,15 @@ import 'css/algolia.css';
 
 class Template extends Component {
   componentDidMount() {
-    // Initialize Algolia search.
-    // TODO Is this expensive? Should it be deferred until a user is about to search?
-    // eslint-disable-next-line no-undef
-    docsearch({
-      apiKey: '36221914cce388c46d0420343e0bb32e',
-      indexName: 'react',
-      inputSelector: '#algolia-doc-search',
+    loadScript(algoliaURL).then(() => {
+      // Initialize Algolia search.
+      // TODO Is this expensive? Should it be deferred until a user is about to search?
+      // eslint-disable-next-line no-undef
+      docsearch({
+        apiKey: '36221914cce388c46d0420343e0bb32e',
+        indexName: 'react',
+        inputSelector: '#algolia-doc-search',
+      });
```

Comment:
This will load the Algolia JS for every page. Even though the request will be cached, the browser might still end up wasting cycles processing/reprocessing the script.

Also, ideally, if we're going to defer loading Algolia- we should defer loading it until the user actually starts to search (eg clicks to focus in the search bar). So maybe we could do this async step there instead? Happy to (a) talk more about this or (b) split that part off into a separate, follow-up issue or task.

Reply:
Perhaps this should be a separate issue.